examples/synchronize_groups_aggregate.py

```diff
@@ -0,0 +1,143 @@
+"""
+Script for synchronizing group(s) between local and aggregate node (for aggregation mode)
+
+Please provide the nodes information in your api.conf file using this format:
+[local_node]
+api_key = ...
+secret_key = ...
+url = https://myinstance.local
+
+[aggregation_node]
+api_key = ...
+secret_key = ...
+url = https://myinstance.aggregate
+"""
+
+import os
+from configparser import ConfigParser
+from urllib.parse import parse_qs
+
+from requests.api import get
+from cbw_api_toolbox.cbw_api import CBWApi
+
+def connect_api(node):
+    '''Connect to the API and test connection'''
+    CONF = ConfigParser()
+    CONF.read(os.path.join(os.path.abspath(
+    os.path.dirname(__file__)), '.', 'api.conf'))
+    client = CBWApi(CONF.get(node, 'url'), CONF.get(
+        node, 'api_key'), CONF.get(node, 'secret_key'))
+    client.ping()
+    return client
+
+def get_groups(client):
+    '''get all groups of an instance by their names'''
+    groups_by_name = []
+    for group in client.groups():
+        groups_by_name.append(group.name)
+    return groups_by_name
+
+def create_groups_on_aggregation_node(groups, client):
+    '''create the missing groups on the aggregation node'''
+    for group in groups:
+        PARAMETERS = {"name": group, "description": "Group from aggregation", "color": "#fa824c"}
+        client.create_group(PARAMETERS)
+
+def get_assets_by_hostname_id(client):
+    '''get all assets of an instance by their hostnames and ids'''
+    assets_by_hostname_id = {}
+    for asset in client.assets():
+        assets_by_hostname_id[asset.hostname] = asset.id
+    return assets_by_hostname_id
+
+def get_asset_groups(node_assets):
+    '''get assiociated groups of an asset'''
+    dict_asset_groups = {}
+    for node_asset in node_assets:
+        groups = []
+        if node_asset.groups:
```

Comment:
We must check at some point that we do not have assets without hostname

Maybe add `node_asset.hostname is not None` to this condition

Or get rid of assets with no hostname directly from the "local_assets" list